### PR TITLE
Fix BFN syncd-rpc compilation

### PR DESCRIPTION
**Pull request by msosyak**
**- What I did**
* Fix BFN syncd-rpc compilation 
* Enable saithriftv2 for BFN syncd-rpc 
* Add docker ptf under bfn platfrom with needed dependencies 

**- How I did it**
Fix rules

**- How to verify it**
This PR relay on https://github.com/barefootnetworks/SAI/pull/2
```
NOJESSIE=1 NOSTRETCH=1 make configure platform=barefoot
ENABLE_SYNCD_RPC=y NOJESSIE=1 NOSTRETCH=1 make target/docker-syncd-bfn-rpc.gz
 
ENABLE_SYNCD_RPC=y NOJESSIE=1 NOSTRETCH=1 make target/docker-ptf.gz
```

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: rules/config [hunk 1]
```diff
@@ -76,7 +76,12 @@ SONIC_ROUTING_STACK = frr
 
 # ENABLE_SYNCD_RPC - build docker-syncd with rpc packages for testing purposes.
 # Uncomment to enable:
-# ENABLE_SYNCD_RPC = y
+ENABLE_SYNCD_RPC = y
```
Comment (akokhan) [inline rules/config]: Do we need to keep this enabled by default? Same below.

Comment (msosyak) [reply rules/config]: Fixed

File: platform/barefoot/libsaithrift-dev.mk [hunk 2]
```diff
@@ -2,19 +2,19 @@
 
 SAI_VER = 0.9.4
 
-LIBSAITHRIFT_DEV = libsaithrift-dev_$(SAI_VER)_amd64.deb
+LIBSAITHRIFT_DEV = libsaithrift-dev_$(SAI_VER)_$(CONFIGURED_ARCH).deb
 $(LIBSAITHRIFT_DEV)_SRC_PATH = $(SRC_PATH)/sonic-sairedis/SAI
-$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT) $(LIBTHRIFT_DEV) $(PYTHON_THRIFT) $(THRIFT_COMPILER) $(BFN_SAI) $(BFN_SAI_DEV)
-$(LIBSAITHRIFT_DEV)_RDEPENDS += $(LIBTHRIFT) $(BFN_SAI)
+$(LIBSAITHRIFT_DEV)_DEPENDS += $(LIBTHRIFT_0_13_0) $(LIBTHRIFT_DEV_0_13_0) $(PYTHON_THRIFT_0_13_0) $(THRIFT_COMPILER_0_13_0) $(LIBSAIMETADATA_DEV) #$(LIBSAIVS) $(LIBSAIVS_DEV) $(LIBSAIMETADATA)
```
Comment (ravi861) [inline platform/barefoot/libsaithrift-dev.mk]: Is there a dependency on thrift 0.13 for the SAI PTF?

File: slave.mk [hunk 3]
```diff
@@ -257,6 +260,7 @@ $(info "FRR_USER_UID"                    : "$(FRR_USER_UID)")
 $(info "FRR_USER_GID"                    : "$(FRR_USER_GID)")
 endif
 $(info "ENABLE_SYNCD_RPC"                : "$(ENABLE_SYNCD_RPC)")
+$(info "ENABLE_SYNCD_RPC_V2"             : "$(ENABLE_SYNCD_RPC_V2)")
```
Comment (ravi861) [inline slave.mk]: Since all of this for the customer only, can we just stick with existing ENABLE_SYNCD_RPC? I don't want to confuse them.

Comment (msosyak) [reply slave.mk]: Yes, I will remove this new option

File: platform/barefoot/libsaithrift-dev.mk (hunk 2 above)
Comment (msosyak) [reply platform/barefoot/libsaithrift-dev.mk]: Actually, thrift 0.13.0 is backward compatible with 0.11.0 in almost all cases, but the original thrift target is built with an additional option [`-DFORCE_BOOST_SMART_PTR` ](https://github.com/Azure/sonic-buildimage/blob/master/src/thrift/Makefile#L30). This causes issues for BFN SDE as we use thrift in driver, platform, and also, in SAIv2 rpc server and everywhere with  `std::smart_ptr`. To fix those issues I have added thrift_0_13_0 target to sonic-buildimage recently https://github.com/Azure/sonic-buildimage/pull/8307. That is why `LIBSAITHRIFT_DEV` depends on `LIBTHRIFT_0_13_0`


UPD: `-DFORCE_BOOST_SMART_PTR` was removed in thrift 0.13.0 and `std::smart_ptr` is used by default.

**Comment by msosyak** [comment]
Closed in favor of https://github.com/Azure/sonic-buildimage/pull/9884

**Pull request closed without merging**